tests/language-feature/capability/var-capability-incompatible.slang

```diff
@@ -0,0 +1,22 @@
+//TEST:SIMPLE(filecheck=CHECK1):-target spirv
+//TEST:SIMPLE(filecheck=CHECK2):-target spirv
+
+[require(hlsl)]
+struct MyType
+{}
+
+//CHECK1: error 36107: entrypoint 'f1' uses features that are not available in 'compute' stage for 'spirv' target.
+[numthreads(1,1,1)]
+void f1()
+{
+    MyType t;  // compile to spirv should result error here.
+}
+
+[numthreads(1,1,1)]
+void f2(uniform ConstantBuffer<MyType> t)  // compile to spirv should result error here.
+{
+     
+}
+
+//CHECK2: error 36107: entrypoint 'f2' uses features that are not available in 'compute' stage for 'spirv' target.
+ConstantBuffer<MyType> t2; // error here too.
```

Comment:
we should only error here when `t2` is referenced by an entrypoint. So I would add some uses of `t2` from `f2`.
If `t2` is not referenced anywhere, it's fine to leave it there without any errors.